src/lib/SnackBar/index.ts

```diff
@@ -0,0 +1,98 @@
+import './styles.css';
+
+const DEFAULT_TIMEOUT = 2750;
+
+export interface SnackBarOptions {
+  message: string;
+  timeout?: number;
+  actionText?: string;
+  actionHandler?: () => boolean | null;
+}
+
+class SnackBar {
+  on?: () => void;
+  onhide?: () => void;
+  private options: SnackBarOptions;
+  private element: Element = document.createElement('div');
+  private text: Element = document.createElement('div');
+  private button: Element = document.createElement('button');
+  private parent: Element;
+  private showing = false;
+  private closeTimer?: number;
+
+  constructor (options: SnackBarOptions, parent: Element) {
+    this.options = options;
+    this.parent = parent;
+
+    this.element.className = 'snackbar';
+    this.element.setAttribute('aria-live', 'assertive');
+    this.element.setAttribute('aria-atomic', 'true');
+    this.element.setAttribute('aria-hidden', 'true');
+
+    this.text.className = 'snackbar--text';
+    this.text.textContent = options.message;
+    this.element.appendChild(this.text);
+
+    if (options.actionText) {
+      this.button.className = 'snackbar--button';
+      this.button.textContent = options.actionText || '';
+      this.button.addEventListener('click', (event) => {
+        if (options.actionHandler && options.actionHandler() === false) return;
+        this.hide();
+      });
+      this.element.appendChild(this.button);
+    }
+  }
+
+  cancelTimer () {
+    if (this.closeTimer != null) clearTimeout(this.closeTimer);
+  }
+
+  show () {
+    if (this.showing) return;
+    this.showing = true;
+    this.cancelTimer();
+    this.parent.appendChild(this.element);
+    this.element.removeAttribute('aria-hidden');
+    this.closeTimer = setTimeout(this.hide.bind(this), this.options.timeout || DEFAULT_TIMEOUT);
+  }
+
+  hide () {
+    if (!this.showing) return;
+    this.showing = false;
+    this.cancelTimer();
+    this.element.addEventListener('animationend', this.remove.bind(this));
+    this.element.addEventListener('transitionend', this.remove.bind(this));
+    this.element.setAttribute('aria-hidden', 'true');
+  }
+
+  remove () {
+    this.cancelTimer();
+    this.parent.removeChild(this.element);
+    if (this.onhide) this.onhide();
+  }
+}
+
+export default class SnackBarElement extends HTMLElement {
```

Comment:
Kinda wondering... why is this a Custom Element? Seems like it should mostly be a module with a `showSnackbar()` function?

Reply:
Yeah, as it stands it doesn't make sense as a custom element. It would make more sense if the snack bar items appeared within the custom element, but if that isn't useful a simple module/class could work.

Reply:
I'd originally had the CE provide a root. Do we think that might be useful? If so it would justify the CE.